acceptance/detector_test.go

```diff
@@ -121,6 +122,61 @@ func testDetector(t *testing.T, when spec.G, it spec.S) {
 		})
 	})
 
+	when("assets env var is set on the builder", func() {
+		var copyDir, containerName string
+		it.Before(func() {
+			containerName = "test-container-" + h.RandString(10)
+			var err error
+			copyDir, err = ioutil.TempDir("", "test-docker-copy-")
+			h.AssertNil(t, err)
+		})
+		it.After(func() {
+			if h.DockerContainerExists(t, containerName) {
+				h.Run(t, exec.Command("docker", "rm", containerName))
+			}
+			os.RemoveAll(copyDir)
+		})
+
+		it.Focus("does not appear in the detection environment", func() {
+			h.DockerRunAndCopy(t,
```

Comment:
You could just use `h.DockerRun` if you're okay with https://github.com/buildpacks/lifecycle/pull/591/files/2ab4678bd230c0aeb6d0c081991f992b0d3953fc..5ff86130c9a0a0f5176c617849564ecbe62ffb4e?file-filters%5B%5D=No+extension#r630437959